### PR TITLE
fix(duckdb): fix tablecolumn for filter following join

### PR DESCRIPTION
**Pull request by gforsyth**
For a join followed by a filter, sqlalchemy backends follow a separate code-path if there are overlapping column names in the joined tables vs. if they are not.

`postgres` also currently fails this test, so I'm not sure why the `ops.TableColumn` in place there is where it is -- going to investigate that more tomorrow. Also need a tinier join than what's here so `pandas` and `dask` don't freak out.

`clickhouse` has one of the year columns as `int64` and the other as `int32` which is causing the failure there (which is annoying).

### CODE REVIEW

**Comment by cpcloud** [comment]
Looks like impala is xpassing!

**Comment by cpcloud** [comment]
You may need to do a `poetry lock --no-update` to appease the dependency gods.

**Comment by cpcloud** [comment]
Oh, it looks like maybe you need to rebase on master

File: ibis/backends/duckdb/registry.py [hunk 1]
```diff
@@ -108,18 +109,19 @@ def _struct_field(t, expr):
...
+        ops.Log2: unary(sa.func.log2),
+        ops.Log: _log,
         # TODO: map operations, but DuckDB's maps are multimaps
+        ops.Modulus: _mod,
+        ops.Round: _round,
         ops.StructField: _struct_field,
+        ops.TableColumn: _table_column,
+        ops.TimestampDiff: fixed_arity('age', 2),
+        ops.TimestampFromUNIX: _timestamp_from_unix,
+        ops.Translate: fixed_arity('replace', 3),
```
Comment (cpcloud) [inline ibis/backends/duckdb/registry.py:125]: Is everything except the addition of the newly imported `_table_column` limited to sorting by op name?

Comment (gforsyth) [reply ibis/backends/duckdb/registry.py:125]: Not sure I follow you -- I did sort the list of op names in here for easier scanning (by human eyes)

File: ibis/backends/postgres/registry.py [hunk 2]
```diff
@@ -524,7 +524,23 @@ def _table_column(t, expr):
...
+        # happens when a join expression is cached during join traversal
+        #
+        # We'd like to avoid generating a subquery just for selection but in
+        # sqlalchemy the Join object is not selectable. However, at this point
+        # know that the column can be referred to unambiguously
+        #
+        # Later the expression is assembled into
+        # `sa.select([sa.column(colname)]).select_from(table_set)` (roughly)
+        # where `table_set` is `sa_table` above.
+        out_expr = sa.column(op.name)
```
[the remark this answers is not in the document]
Comment (gforsyth) [reply ibis/backends/postgres/registry.py:543]: yeah, that's a good idea

File: ibis/backends/duckdb/registry.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/duckdb/registry.py:125]: Sorry, my hands didn't do what my brain was thinking. What I meant to ask was:

Is every change here except the addition of the newly imported `_table_column` limited to sorting by op name?

File: ibis/backends/tests/test_join.py [hunk 3]
```diff
@@ -175,3 +175,24 @@ def test_filtering_join(backend, con, batting, awards_players, how):
...
+@pytest.mark.notimpl(["datafusion", "impala"])
+def test_join_then_filter_no_column_overlap(awards_players, alltypes):
+
+    expr = awards_players.join(
+        alltypes, awards_players.yearID == alltypes.year
+    )
+    filters = [expr.playerID == "renteed01"]
+    q = expr.filter(filters)
+    q.limit(10)
+    q.execute()
```
[the remark this answers is not in the document]
Comment (gforsyth) [reply ibis/backends/tests/test_join.py:198]: I'm not sure -- the test is whether `execute` can, well, execute.  We can add an assertion but in some ways I think it'll distract from the test's purpose.

File: ibis/backends/duckdb/registry.py (hunk 1 above)
Comment (gforsyth) [reply ibis/backends/duckdb/registry.py:125]: yes, correct

**Review by cpcloud — APPROVED** [review]
LGTM! If you can squash as appropriate, I'll merge.

File: ibis/backends/tests/test_join.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/tests/test_join.py:198]: Sounds good.

File: pyproject.toml [hunk 4]
```diff
@@ -85,6 +85,7 @@ requests = ">=2,<3"
 setuptools = ">=57,<61"
 sqlalchemy = ">=1.3,<1.5"
 types-requests = ">=2.27.8,<3"
+ipdb = "^0.13.9"
```
Comment (cpcloud) [inline pyproject.toml:88]: Would you mind sorting this?

**Pull request merged**